src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/_params.py

```diff
@@ -33,6 +33,7 @@
 register_cli_argument('batch account create', 'location', location_type, help='The region in which to create the account.')
 register_cli_argument('batch account create', 'tags', tags_type, help="Space separated tags in 'key[=value]' format.")
 register_extra_cli_argument('batch account create', 'storage_account', help='The storage account name or resource ID to be used for auto storage.', validator=storage_account_id)
+register_extra_cli_argument('batch account create', 'keyvault', help='The keyvault name or resource ID to be used for an account with a pool allocation mode of \'User Subscription\'.', validator=keyvault_id)
```

Comment:
@xingwu1 - I've only added one additional argument '--keyvault' and made the assumption that if this value is supplied, the created account will always be 'UserSubscription', otherwise it will always be 'BatchService'.
Do you think this behavior is okay?